shapely/tests/test_linear.py

```diff
@@ -136,6 +137,22 @@ def test_line_merge_geom_array():
     assert_geometries_equal(actual[1], LineString([(0, 0), (1, 2)]))
 
 
+@pytest.mark.skipif(shapely.geos_version < (3, 11, 0), reason="GEOS < 3.11.0")
+def test_line_merge_directed():
+    lines = MultiLineString([[(0, 0), (1, 0)], [(0, 0), (3, 0)]])
+    result = shapely.line_merge(lines)
+    assert_geometries_equal(result, LineString([(1, 0), (0, 0), (3, 0)]))
+    result = shapely.line_merge(lines, directed=True)
+    assert_geometries_equal(result, lines)
```

Comment:
What happens if it is not possible to merge the lines (without changing point order)? Does GEOS raise an exception? Can we add a test for that as well?

Reply:
This is what the test is showing, that `result` is the same as the input `lines`.

Reply:
The input is returned if it can't merge the lines.

Reply:
Ah, yes, that's indeed what the test is showing here. I didn't get that from the docstring. I would maybe explicitly mention that?

Reply:
Now, the fact that it returns the input unchanged if it was not possible to combine them, that's actually also the behaviour of the original (non-directed) `line_merge` function, so nothing changed here. 
So we don't need to specify this specifically for `directed`, but maybe it might still be useful to mention in general.